pkg/planner/core/logical_plan_builder.go

```diff
@@ -5613,6 +5613,9 @@ func (b *PlanBuilder) buildUpdateLists(ctx context.Context, tableList []*ast.Tab
 					break
 				}
 			}
+			if isModified {
+				dependentColumnsModified[col.UniqueID] = true
```

Comment:
Why not put this line to line 5612?

Reply:
The col in L5612 is not the col in L5617.